### PR TITLE
WIP experiment with standalone Nimbus iOS XCFramework [ci skip]

### PR DESCRIPTION
**Pull request by rfk**
(Draft PR for early visibility)

Leaning heavily on @badboy's excellent work on [building Glean as an XCFramework](https://github.com/mozilla/glean/pull/1663), I've started experimenting with what it would look like to do the same for Nimbus. My short-term goal is to see if I can make a Nimbus equivalent of the [glean-swift repo](https://github.com/mozilla/glean-swift) and try consuming Nimbus as a Swift Package. Not because we have consumers that want to use it that way yet, but to help me get my head around all the moving parts.

Connects to #4212.

### CODE REVIEW

**Comment by rfk** [comment]
One early observation: I was initially hoping to do this for *just* Nimbus, but I think we have to do it for a little Nimbus + Glean megazord, because the Nimbus Swift bindings directly use the Glean Swift bindings, and I don't know how well it would work to have Nimbus instead try to depend on the Glean Swift Package. We'll see...

File: build-scripts/xc-universal-er-binary.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
```
Comment (rfk) [inline build-scripts/xc-universal-er-binary.sh:1]: This script is also ~shamelessly copied from~ based heavily on the work done to compile Glean for the M1. We probably want to push the changes here up into our existing `xc-universal-binary.sh` script, even without any of the new XCFramework/Swift-Package stuff.

File: megazords/ios/MozillaAppServices.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -3,7 +3,7 @@
 	archiveVersion = 1;
 	classes = {
 	};
-	objectVersion = 52;
+	objectVersion = 51;
```
Comment (rfk) [inline megazords/ios/MozillaAppServices.xcodeproj/project.pbxproj:6]: Eh? XCode doing XCode things maybe?

File: megazords/nimbus-ios/NimbusStandalone.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -0,0 +1,587 @@
+// !$*UTF8*$!
```
Comment (rfk) [inline megazords/nimbus-ios/NimbusStandalone.xcodeproj/project.pbxproj:1]: This and its supporting `.swift` files were copied from the iOS megazord and heavily trimmed down. Depending on how the build process shapes up, we might want to consider moving e.g. the Swift RustLog and Viaduct stuff out of the iOS megazord and into some shared location.

**Comment by st3fan** [comment]
@rfk This is awesome progress. I'll be the first consumer with Focus iOS. Let me know if you have either a swift package or just the xcframework and i'll integrate that into a test branch of Focus.

**Comment by rfk** [comment]
Some more progress here:

* I've made https://github.com/rfk/nimbus-swift as a swift package that publishes the pre-built Nimbus+Glean binary artifact
* I've hacked up focus-ios so that it pulls this as a dependency rather than the standalone glean, ref https://github.com/rfk/focus-ios/pull/1
* I've further hacked up focus-ios to make some calls into the Nimbus API
* When I build this hacked-up focus-ios, it seems to run successfully and to actually make calls to Nimbus!

focus-nimbus-hackery

From this I interpret that the XCFramework stuff is actually compiling correctly. @st3fan if you're happy to check out the above and see if it seems sensible, that would be great.

Next, I need to see if I can understand what it would look like for Nimbus to consume Glean as a swift package, rather than requiring consumers like Focus to know about megazording. Megazording may turn out to be the only tractable approach in the short term, but it'll be worth understanding the landscape.

**Comment by rfk** [comment]
This was a really useful experiment, but I'm going to close it in favour of something *even better* which I'll share more about shortly 😁

**Pull request closed without merging**